app/models/offer.rb

```diff
@@ -1,4 +1,12 @@
 class Offer < ApplicationRecord
   belongs_to :application_choice
   has_many :conditions, class_name: 'OfferCondition'
+
+  def mark_all_conditions_as_met!
+    conditions.update(status: :met)
```

Comment:
```suggestion
    conditions.update!(status: :met)
```

Reply:
`update!` is not a method on a collection. I'm doing this instead of `update_all` because `update_all` skips callbacks (and hence auditing)

Reply:
I think that would be best, and then the method name would also correctly reflect that failure causes an exception to be thrown.